esp-hal-embassy/src/time_driver.rs

```diff
@@ -203,11 +203,10 @@ impl EmbassyTimer {
     /// Returns `true` if the timer was armed, `false` if the timestamp is in
     /// the past.
     fn arm(timer: &mut Timer, timestamp: u64) -> bool {
-        let now = now().duration_since_epoch();
-        let ts = timestamp.micros();
+        let now = now().micros_since_epoch();
```

Comment:
I wonder if we're going with our own types we should also make `Instant::now` a thing instead of `esp_hal::time::now`? I suppose if we do this we won't be able to move the time types out of the HAL though 🤔

Reply:
Instant is meaningless without the HAL, you can't create an Instant anyway. That the underlying number is `micros since epoch` is an implementation detail that this PR hides from pretty much everything.

So I'm not sure if we want to move these out of the hal, or what we would gain by doing that. At present both `Instant::now` and `esp_hal::time::now` exist so that we don't break even more things.

Reply:
> So I'm not sure if we want to move these out of the hal, or what we would gain by doing that

I was mostly referring to the part about adding conversion impls for various crates from the parent issue, like embassy-time in the future, but I'm not really convinced we need to do that. It should be fine to live here in the HAL.

Reply:
We can always add an `embassy_time_04` feature (if https://github.com/esp-rs/esp-hal/issues/3063 goes that way) to esp-hal.